flaml/automl.py

```diff
@@ -183,20 +183,24 @@ def __init__(
                 isinstance(starting_point, dict)
                 and starting_point.get(name) is not None
             ):
+                if self.init_config is None:
+                    self.init_config = {}
```

Comment:
Would it be cleaner to move the evaluation of `self.init_config` to before the for-loop? Looks like doing it once before starting the iteration would have the same result.

Reply:
There is a chance that the evaluation of `self.init_config` would not be evaluated at all in the for-loop. If we move it out, we would be setting self.init_config to {} no matter what. That's what I wanted to avoid.

Reply:
I see. Feels like the logic could benefit from some refactoring. However, that may very well be best kept to a separate PR.